libpromises/processes_select.c

```diff
@@ -548,6 +548,29 @@ static bool SelectProcRegexMatch(const char *name1, const char *name2,
     return false;
 }
 
+static void PrintStringIndexLine(int prefix_spaces, int len)
+{
+    char arrow_str[CF_BUFSIZE];
+    arrow_str[0] = '^';
+    arrow_str[1] = '\0';
+    char index_str[CF_BUFSIZE];
+    index_str[0] = '0';
+    index_str[1] = '\0';
+    for (int lineindex = 10; lineindex <= len; lineindex += 10)
+    {
+        char num[11];
+        xsnprintf(num, sizeof(num), "% 10d", lineindex);
```

Comment:
It's perhaps better explained by demonstration. The section will produce debug output like this:

```
   debug: Parsing ps line: 'root         1     0     1  0.0  0.0  24728   0      2416    1 Apr12  2-03:59:55 00:00:01 /sbin/init'
   debug:                   ^         ^         ^         ^         ^         ^         ^         ^         ^         ^         ^
   debug:            Index: 0        10        20        30        40        50        60        70        80        90       100
```